src/pages/SettingsPage.tsx

```diff
@@ -1,7 +1,189 @@
-import React from "react";
+import React, { useState } from "react";
+import {
+  Radio,
+  RadioGroup,
+  FormControlLabel,
+  FormControl,
+  FormLabel,
+  Box,
+  Grid,
+  InputAdornment,
+  IconButton,
+  Input,
+  Button,
+} from "@material-ui/core";
+import { Backspace, LocalPhone } from "@material-ui/icons";
 import { Header } from "components/Header";
 import { labels } from "labels";
 
+interface Options {
+  value: string;
+  label: string;
+}
+interface SettingsRadioGroupProps {
+  formLabel: string;
+  name: string;
+  options: Options[];
+  value: string;
+  setValue: (to: string) => void;
+}
+interface SettingKeyPadProps {
+  buttonsValues: string[];
+  number: string;
+  setNumber: (newNum: any) => void;
+}
+const SettingsRadioGroup: React.FC<SettingsRadioGroupProps> = ({
+  name,
+  formLabel,
+  options,
+  value,
+  setValue,
+}) => {
+  return (
+    <FormControl component="fieldset">
+      <FormLabel component="legend">{formLabel}</FormLabel>
+      <RadioGroup
+        name={name}
+        value={value}
+        onChange={(e) => setValue(e.target.value)}
+      >
+        {options.map((option, index) => (
+          <FormControlLabel
+            key={index}
+            value={option.value}
+            control={<Radio color="primary" />}
+            label={option.label}
+          />
+        ))}
+      </RadioGroup>
+    </FormControl>
+  );
+};
+
+const SettingKeyPad: React.FC<SettingKeyPadProps> = ({
+  buttonsValues,
+  number,
+  setNumber,
+}) => {
+  return (
+    <Grid container spacing={0}>
+      <Grid item xs={12}>
```

Comment:
I think this component should be renamed to `Dialpad` and should get split up into three separate subcomponents. `SearchBar` & `Keypad` & `CallButton`